src/client/WindshaftWorker.js

```diff
@@ -0,0 +1,30 @@
+import { MVTWorker } from '../sources/MVTWorker';
+import schema from '../renderer/schema';
+
+export class WindshaftWorker extends MVTWorker {
+    decodeProperty (metadata, propertyName, propertyValue) {
+        const basename = schema.column.getBase(propertyName);
+        const column = metadata.properties[basename];
+        if (!column) {
+            return;
+        }
+        switch (column.type) {
+            case 'date':
+                return decodeDate(column, propertyValue);
+            case 'category':
+                return metadata.categorizeString(basename, propertyValue);
+            case 'number':
+                return propertyValue;
+            default:
+                throw new Error(`Windshaft MVT decoding error. Feature property value of type '${typeof propertyValue}' cannot be decoded.`);
+        }
+    }
+}
+
+function decodeDate (column, propertyValue) {
```

Comment:
nice refactor. all this is actually better now, being encapsulated in a subclass